Content.Server/Physics/Controllers/ChasingWalkSystem.cs

```diff
@@ -0,0 +1,116 @@
+        component.NextChangeVectorTime = _gameTiming.CurTime;
+    }
+
+    private void OnChasingUnpaused(EntityUid uid, ChasingWalkComponent component, ref EntityUnpausedEvent args)
+    {
+        component.NextImpulseTime += args.PausedTime;
+        component.NextChangeVectorTime += args.PausedTime;
+    }
+
+    public override void UpdateBeforeSolve(bool prediction, float frameTime)
+    {
+        base.UpdateBeforeSolve(prediction, frameTime);
+
+        var query = EntityQueryEnumerator<ChasingWalkComponent>();
+        while (query.MoveNext(out var uid, out var chasing))
+        {
+            //Set Velocity to Target
+            if (chasing.NextImpulseTime <= _gameTiming.CurTime)
+            {
+                ForceImpulse(uid, chasing);
+                chasing.NextImpulseTime += TimeSpan.FromSeconds(chasing.ImpulseInterval);
+            }
+            //Change Target
+            if (chasing.NextChangeVectorTime <= _gameTiming.CurTime)
+            {
+                ChangeTarget(uid, chasing);
+
+                var delay = TimeSpan.FromSeconds(_random.NextFloat(chasing.ChangeVectorMinInterval, chasing.ChangeVectorMaxInterval));
+                chasing.NextChangeVectorTime += delay;
+            }
+        }
+    }
+
+    private void ChangeTarget(EntityUid uid, ChasingWalkComponent component)
+    {
+        //We find our coordinates and calculate the radius of the target search.
+        var xform = Transform(uid);
+        var range = component.MaxChaseRadius;
+        var compType = _random.Pick(component.ChasingComponent.Values).Component.GetType();
+        _potentialChaseTargets.Clear();
+        _lookup.GetEntitiesInRange(compType, _transform.GetMapCoordinates(xform), range, _potentialChaseTargets, LookupFlags.Uncontained);
+
+        //If there are no required components in the radius, don't moving.
+        if (_potentialChaseTargets.Count <= 0)
+            return;
+
+        //In the case of finding required components, we choose a random one of them and remember its uid.
+        component.ChasingEntity = _random.Pick(_potentialChaseTargets).Owner;
+        component.Speed = _random.NextFloat(component.MinSpeed, component.MaxSpeed);
+    }
+
+    //pushing the entity toward its target
+    private void ForceImpulse(EntityUid uid, ChasingWalkComponent component)
+    {
+        if (Deleted(component.ChasingEntity) || component.ChasingEntity == null)
+        {
+            ChangeTarget(uid, component);
+            return;
+        }
+
+        if (!TryComp<PhysicsComponent>(uid, out var physics))
+            return;
+
+        //Calculating direction to the target.
+        var xform = Transform(component.ChasingEntity.Value);
+
+        if (xform.Coordinates.TryDelta(EntityManager, _transform, _xformQuery.Get(uid).Comp.Coordinates, out var delta))
+        {
+            //Changing the direction of the entity.
+            var speed = delta.Length() > 0 ? delta.Normalized() * component.Speed : Vector2.Zero;
```

Comment:
For some reason now micro lightning bolts have a chance to go somewhere completely different than the tesla.